packages/analytics/analytics-chart/src/components/chart-plugins/ChartLegend.vue

```diff
@@ -140,11 +141,26 @@ watch(() => props.items, () => {
   }
 }, { immediate: true, flush: 'post' })
 
-const resizeObserver = new ResizeObserver(() => {
-  checkForWrap()
-})
+const oldWidth = ref(0)
+const oldHeight = ref(0)
+
+const resizeObserver = new ResizeObserver(debounce((entries: ResizeObserverEntry[]) => {
+
+  const entry = entries[0]
+  const newWidth = entry.contentRect.width
+  const newHeight = entry.contentRect.height
+
+  // Check if the resize is only in the X direction
+  if (newWidth !== oldWidth.value && newHeight === oldHeight.value) {
```

Comment:
somewhat minor, but if don't need to `checkForWrap()` or `formatGrid()` on the very first page load, then we could exclude `oldWidth.value === 0`

Reply:
we do want on the first page load :)